docs/eloquent-models/model-class.txt

```diff
@@ -0,0 +1,292 @@
+MongoDB data by using the Laravel Eloquent object-relational mapper (ORM).
+
+The following sections explain how to add Laravel Eloquent ORM behaviors
+to {+odm-short+} models:
+
+- :ref:`laravel-model-define` demonstrates how to create a model class.
+- :ref:`laravel-authenticatable-model` shows how to set MongoDB as the
+  authentication user provider.
+- :ref:`laravel-model-customize` explains several model class customizations.
+- :ref:`laravel-model-pruning` shows how to periodically remove models that
+  you no longer need.
+
+.. _laravel-model-define:
+
+Define an Eloquent Model Class
+------------------------------
+
+Eloquent models are classes that represent your data. They include methods
+that perform database operations such as inserts, updates, and deletes.
+
+To declare a {+odm-short+} model class, create a file in your Laravel
+application ``app/Models`` directory. Add the import for the
+``MongoDB\Laravel\Eloquent\Model`` package and extend the ``Model`` class as
+shown in the following code example:
+
+.. literalinclude:: /includes/eloquent-models/Planet.php
+   :language: php
+   :dedent:
+
+Alternatively, use the ``artisan`` console to generate the model class and
+change the ``Illuminate\Database\Eloquent\Model`` import to ``MongoDB\Laravel\Eloquent\Model``.
+To learn more about the ``artisan`` console, see `Artisan Console <https://laravel.com/docs/{+laravel-docs-version+}/artisan>`__
+in the Laravel docs.
+
+.. _laravel-authenticatable-model:
+
+Extend the Authenticatable Model
+--------------------------------
+
+To configure MongoDB as the Laravel user provider, you can extend the
+{+odm-short+} ``MongoDB\Laravel\Auth\User`` class. The following code example
+shows how to extend this class:
+
+.. literalinclude:: /includes/eloquent-models/AuthenticatableUser.php
+   :language: php
+   :dedent:
+
+To learn more about customizing a Laravel authentication user provider,
+see `Adding Custom User Providers <https://laravel.com/docs/{+laravel-docs-version+}/authentication#adding-custom-user-providers>`__
+in the Laravel docs.
+
+.. _laravel-model-customize:
+
+Customize an Eloquent Model Class
+---------------------------------
+
+This section shows how to perform the following Eloquent model behavior
+customizations:
+
+- :ref:`laravel-model-customize-collection-name`
+- :ref:`laravel-model-customize-primary-key`
+- :ref:`laravel-model-soft-delete`
+- :ref:`laravel-model-cast-data-types`
+- :ref:`laravel-model-mass-assignment`
+
+.. _laravel-model-customize-collection-name:
+
+Change the Model Collection Name
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To customize the name of the collection the model uses to retrieve and save
+data in MongoDB, override the ``$collection`` property of the model
+class.
+
+By default, the model uses the MongoDB database name set in your Laravel
+application's ``config/database.php`` setting and the snake case plural form of
+your model class name for the collection.
```

Comment:
I find it confusion. You should tell the default value first. And tell the collection name *can* be modified, but only if necessary and this is not recommended.

Why do you mention the database here?

Reply:
> I find it confusion. You should tell the default value first. And tell the collection name _can_ be modified, but only if necessary and this is not recommended.

Thanks for identifying that. I was debating between placing this paragraph in the prior section "Define an Eloquent Model Class" because it could be useful to know. I think the changes I'll make the following changes based on your feedback:

- Add mention of the default collection name in "Define an Eloquent Model Class"
- In this section, restate the default collection, recommend not modifying it, and provide the example as mentioned in your next comment.

> Why do you mention the database here?

I think this information hasn't been presented anywhere else and thought it could be helpful information for users that are attempting to locate their app data in MongoDB. Let me know if you have other ideas for this.

Reply:
The proposed change seems good.

The database name should be in a page about the configuration. Like this: https://github.com/mongodb/laravel-mongodb/blob/4.0/docs/install.md#configuration

Reply:
Quick question:
I'd like to provide a reason why they shouldn't customize it. Would it be correct to say "we recommend using the default collection naming behavior to prevent issues such as collection name collisions"? Otherwise, what would be a better reason to call out?

Reply:
Keeping the default name will make the project more "standard". This will help developers to know which collection is associated with which model.

Reply:
> The proposed change seems good.
> 
> The database name should be in a page about the configuration. Like this: https://github.com/mongodb/laravel-mongodb/blob/4.0/docs/install.md#configuration

Thanks -- for now, I'll link to the related [Quick Start step](https://www.mongodb.com/docs/drivers/php/laravel-mongodb/current/quick-start/configure-mongodb/), and will update this in https://jira.mongodb.org/browse/DOCSP-35938.